protocol/linkpreview_unfurler_status.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/status-im/status-go/protocol/communities"
+)
+
+type StatusUnfurler struct {
+	m      *Messenger
+	logger *zap.Logger
+	url    string
+}
+
+func NewStatusUnfurler(URL string, messenger *Messenger, logger *zap.Logger) *StatusUnfurler {
+	return &StatusUnfurler{
+		m:      messenger,
+		logger: logger,
+		url:    URL,
+	}
+}
+
+func buildThumbnail(image *images.IdentityImage, thumbnail *common.LinkPreviewThumbnail) error {
+	if image.IsEmpty() {
+		return nil
+	}
+
+	var err error
+
+	thumbnail.Width, thumbnail.Height, err = images.GetImageDimensions(image.Payload)
+	if err != nil {
+		return fmt.Errorf("failed to get image dimensions: %w", err)
+	}
+
+	thumbnail.DataURI, err = image.GetDataURI()
+	if err != nil {
+		return fmt.Errorf("failed to get data uri: %w", err)
+	}
+
+	return nil
+}
+
+func (u *StatusUnfurler) buildContactData(contactData *ContactURLData) (*common.StatusContactLinkPreview, error) {
+	c := new(common.StatusContactLinkPreview)
+	c.PublicKey = contactData.PublicKey
+	c.DisplayName = contactData.DisplayName
+	c.Description = contactData.Description
+
+	contactID, err := multiformat.DeserializeCompressedKey(contactData.PublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	contact := u.m.GetContactByID(contactID)
+	if contact == nil {
+		return c, nil
+	}
+
+	// TODO: Should we try to fetch from waku?
+	//if contact == nil {
+	//	if contact, err = u.m.RequestContactInfoFromMailserver(contactData.PublicKey, true); err != nil {
+	//		u.logger.Warn("StatusUnfurler: failed to request contact info from mailserver")
+	//		return c, err
+	//	}
+	//}
+
+	if image, ok := contact.Images[images.SmallDimName]; ok {
+		if err = buildThumbnail(&image, &c.Icon); err != nil {
+			return c, fmt.Errorf("failed to set thumbnail: %w", err)
+		}
+	}
+
+	return c, nil
+}
+
+func (u *StatusUnfurler) fillCommunityImages(community *communities.Community, icon *common.LinkPreviewThumbnail, banner *common.LinkPreviewThumbnail) error {
+	if image, ok := community.Images()[images.SmallDimName]; ok {
+		if err := buildThumbnail(&images.IdentityImage{Payload: image.Payload}, icon); err != nil {
+			u.logger.Warn("unfurling status link: failed to set community thumbnail", zap.Error(err))
+		}
+	}
+
+	if image, ok := community.Images()[images.BannerIdentityName]; ok {
+		if err := buildThumbnail(&images.IdentityImage{Payload: image.Payload}, banner); err != nil {
+			u.logger.Warn("unfurling status link: failed to set community banner", zap.Error(err))
+		}
+	}
+
+	return nil
+}
+
+func (u *StatusUnfurler) buildCommunityData(data *CommunityURLData) (*common.StatusCommunityLinkPreview, error) {
+	c := new(common.StatusCommunityLinkPreview)
+
+	// First, fill the output with the data from URL
+	c.CommunityID = data.CommunityID
+	c.DisplayName = data.DisplayName
+	c.Description = data.Description
+	c.MembersCount = data.MembersCount
+	c.Color = data.Color
+	c.TagIndices = data.TagIndices
+
+	// Now check if there's newer information in the database
```

Comment:
if we get an url with community data it really can be outdated. WDYT about updating all the fields from object?

Reply:
Good question, will discuss all together here: 
https://github.com/status-im/status-go/pull/4033/files/69545bc6e44e27c644c15ed0ef17752d61d77230#r1342465007